sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamAccumulatorProvider.java

```diff
@@ -32,38 +31,66 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Translation of accumulators to {@link Metrics}. Metrics from same type of accumulator will have
- * same namespace and will differ only in given name. */
+/**
+ * Translation of accumulators to {@link Metrics}. Metric's namespace is taken from operatorName.
+ * So for better orientation in metrics it's recommended specify operator name with method .named().
+ */
 public class BeamAccumulatorProvider implements AccumulatorProvider {
 
   private static final Logger LOG = LoggerFactory.getLogger(BeamAccumulatorProvider.class);
-  private final Map<String, Counter> counterMap;
-  private final Map<String, Timer> timerMap;
-  private final Map<String, Histogram> histogramMap;
-
-  private BeamAccumulatorProvider() {
-    counterMap = new ConcurrentHashMap<>();
-    timerMap = new ConcurrentHashMap<>();
-    histogramMap = new ConcurrentHashMap<>();
-  }
+  private static final String KEY_METRIC_SEPARATOR = "::";
+
+  private final Map<String, Counter> counterMap = new ConcurrentHashMap<>();
+  private final Map<String, Histogram> histogramMap = new ConcurrentHashMap<>();
+
+  private BeamAccumulatorProvider() {}
 
   public static Factory getFactory() {
     return Factory.get();
   }
 
   @Override
   public Counter getCounter(String name) {
-    return counterMap.computeIfAbsent(name, BeamMetricsCounter::new);
+    throw new UnsupportedOperationException(
+        "BeamAccumulatorProvider doesn't support"
+            + " getCounter(String name). Need specify operatorName and name");
+  }
+
+  @Override
+  public Counter getCounter(final String operatorName, final String name) {
+    return counterMap.computeIfAbsent(
+        getMetricKey(operatorName, name), key -> new BeamMetricsCounter(operatorName, name));
   }
 
   @Override
   public Histogram getHistogram(String name) {
-    return histogramMap.computeIfAbsent(name, BeamMetricsHistogram::new);
+    throw new UnsupportedOperationException(
+        "BeamAccumulatorProvider doesn't support"
+            + " getHistogram(String name). Need specify operatorName and name");
+  }
+
+  @Override
+  public Histogram getHistogram(final String operatorName, final String name) {
+    return histogramMap.computeIfAbsent(
+        getMetricKey(operatorName, name), key -> new BeamMetricsHistogram(operatorName, name));
   }
 
   @Override
   public Timer getTimer(String name) {
-    return timerMap.computeIfAbsent(name, BeamMetricsTimer::new);
+    throw new UnsupportedOperationException(
+        "BeamAccumulatorProvider doesn't support"
+            + " getTimer(String name). Need specify operatorName and name");
+  }
+
+  @Override
+  public Timer getTimer(String operatorName, String name) {
+    throw new UnsupportedOperationException(
+        "BeamAccumulatorProvider doesn't support"
+            + " getTimer(String name). Use getHistogram instead");
+  }
+
+  private static String getMetricKey(String operatorName, String name) {
```

Comment:
I think we should be consistent, either name the parameter `operatorName` or `namespace`.  I'm 👍 for `namespace`.